<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!-- <xacro:property name="description_package" value="brobot_control"/> -->

     <ros2_control name="GazeboSystem" type="system">
        <hardware>
          <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="joint_1">
            <command_interface name="position">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_vjoint">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_2">
            <command_interface name="position">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_3">
            <command_interface name="position">
                <!-- <param name="min">-3.14</param>
                <param name="max">3.14</param> -->
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_4">
            <command_interface name="position">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
            <robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
            <parameters>$(find brobot_Gazebo)/config/brobot_Gazebo.yaml</parameters>
        </plugin>
    </gazebo>

</robot>